cypress/e2e/buy-shirt.cy.ts

```diff
@@ -1,28 +1,38 @@
+import {MenuContentPage} from "../page/index";
+import {ProductsListPage} from "../page/products-list.page";
+import {ShoppingCartPage} from "../page/shopping-cart.page";
+import {LoginPage} from "../page/login.page";
+import {AddressStepPage} from "../page/address-step.page";
+import {ShippingStepPage} from "../page/shipping-step.page";
+import {PaymentStepPage} from "../page/payment-step.page";
+
+const menuContentPage = new MenuContentPage();
+const productsListPage = new ProductsListPage();
+const shoppingCartPage = new ShoppingCartPage();
+const loginPage = new LoginPage();
+const addressStepPage = new AddressStepPage();
+const shippingStepPage = new ShippingStepPage();
+const paymentStepPage = new PaymentStepPage();
+
 describe("Buy a t-shirt", () => {
   it("then the t-shirt should be bought", () => {
-    cy.visit("http://automationpractice.com/");
-    cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
-    cy.get(
-        "#center_column a.button.ajax_add_to_cart_button.btn.btn-default",
-    ).click();
-    cy.get("[style*='display: block;'] .button-container > a").click();
-    cy.get(".cart_navigation span").click();
-
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
-    cy.get("#SubmitLogin").click();
-
-    cy.get(".cart_navigation span").click();
-
-    cy.get("#cgv").click();
-    cy.get(".cart_navigation span").click();
-
-    cy.get(".bankwire ").click();
-    cy.get(".cart_navigation span").click();
-
-    cy.get("#center_column > div > p > strong").should(
-        "have.text",
-        "Your order on My Store is complete.",
-    );
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+
+    productsListPage.addToCart();
+    productsListPage.proceedToCheckOut();
+
+    shoppingCartPage.proceedToCheckout();
+
+    loginPage.signIn();
+
+    addressStepPage.proceedToCheckout();
+
+    shippingStepPage.agreeTerms();
+    shippingStepPage.proceedToCheckout();
+
+    paymentStepPage.payByBankWire();
+    paymentStepPage.confirmOrder();
+    paymentStepPage.verifyText();
```

Comment:
It's usually better for readability of the test to include the assertions (shoulds and stuff like that) inside the test itself explicitly. So I'd change this for a `page.text().should(...)`